ecr-login/api/factory.go

```diff
@@ -43,6 +44,14 @@ func (defaultClientFactory DefaultClientFactory) NewClient(region string) Client
 	}
 }
 
+// NewClientWithAWSConfig Create new client with AWS Config
```

Comment:
Is there any reason to retain `NewClient`?  If so, can you change the name to `NewClientWithRegion`?

Also, you can simplify very slightly by having `NewClientWithRegion` call `NewClientWithAWSConfig`; that way you don't have two different call paths for `defaultClientFactory.buildCredentialsCache`.